### DISCUSSION TITLE
How can I get API keys

### DISCUSSION THREAD

**Discussion by zhuyue0825** [drakkan/sftpgo#1719] [Q&A]
I'm new. I don't know how to get my API keys.

**Comment by stumpyofpain** [comment]
i had the same problem and it took me quite some time to figure it out.

this will generate the bearer token
curl -u 'admin_name:admin_password' -X GET "http://localhost:8080/api/v2/token"

output:
{"access_token":"some long token","expires_at":"2026-03-15T21:13:25Z"}

with the bearer token we can generate the api key, here for an admin
curl -X POST "http://localhost:8080/api/v2/apikeys" \
     -H "Authorization: Bearer the_generatet_token" \
     -H "Content-Type: application/json" \
     -d '{
       "name": "name_of_the_api_key",
       "scope": 1,
       "admin": "admin_name"
     }'

{"key":"api_key","message":"API key created. This is the only time the API key is visible, please save it."}
